controllers/awsmachine_tags.go

```diff
@@ -62,6 +70,35 @@ func (r *AWSMachineReconciler) ensureTags(svc service.EC2MachineInterface, machi
 	return changed, nil
 }
 
+// Ensure that the tags of the volumes in the machine are correct
+// Returns bool, error
+// Bool indicates if changes were made or not, allowing the caller to decide
+// if the machine should be updated.
+func (r *AWSMachineReconciler) ensureVolumeTags(svc service.EC2MachineInterface, machine *infrav1.AWSMachine, volume *infrav1.Volume, instanceID *string, additionalTags map[string]string) (bool, error) {
+	annotations := make(map[string]interface{})
+	annotations["Volumes"] = volume
+
+	// Check if the volume tags were changed. If they were, update them.
+	// It would be possible here to only send new/updated tags, but for the
+	// moment we send everything, even if only a single tag was created or
+	// updated.
+	changed, created, deleted, newAnnotation := r.tagsChanged(annotations, additionalTags)
+	if changed {
+		err := svc.UpdateResourceTags(instanceID, created, deleted)
```

Comment:
We want to update the tags on the volume, not the EC2 instance, which is handled in the `ensureTags` function. The input to `UpdateResourceTags` should start with the volume ID